src/fhirtypes/StructureDefinition.ts

```diff
@@ -60,11 +62,13 @@ export class StructureDefinition {
    * A base clone of the Structure Definition from before any rules were applied
    */
   private _baseStructureDefinition: StructureDefinition;
+  private _structureDefinitionStructureDefinition: StructureDefinition;
 
   /**
    * Constructs a StructureDefinition with a root element.
+   * @param {StructureDefinition} structureDefinitionStructureDefinition - The StructureDefinition of StructureDefinition
    */
-  constructor() {
+  constructor(structureDefinitionStructureDefinition: StructureDefinition = null) {
```

Comment:
I don't love having to pass in the StructureDefinition definition into the constructor.  It just feels real weird to me.

I think we require the `resolve` function as an argument to any functions that need the structdef anyway, and `resolve` can look up the StructureDefinition in the FHIR defs, so... I'd suggest taking this out of the constructor -- and using `resolve` at the appropriate time/place instead.